src/components/drawing/index.js

```diff
@@ -654,18 +653,41 @@ drawing.bBox = function(node) {
     // make sure we don't have too many saved boxes,
     // or a long session could overload on memory
     // by saving boxes for long-gone elements
-    if(savedBBoxes.length >= maxSavedBBoxes) {
-        d3.selectAll('[data-bb]').attr('data-bb', null);
-        savedBBoxes = [];
+    if(savedBBoxesCount >= maxSavedBBoxes) {
+        savedBBoxes = {};
+        maxSavedBBoxes = 0;
     }
 
     // cache this bbox
-    node.setAttribute('data-bb', savedBBoxes.length);
-    savedBBoxes.push(bb);
+    savedBBoxes[hash] = bb;
+    savedBBoxesCount++;
 
     return Lib.extendFlat({}, bb);
 };
 
+// capture everything about a node (at least in our usage) that
+// impacts its bounding box, given that bBox clears x, y, and transform
+// TODO: is this really everything? Is it worth taking only parts of style,
+// so we can share across more changes (like colors)? I guess we can't strip
+// colors and stuff from inside innerHTML so maybe not worth bothering outside.
+// TODO # 2: this can be long, so could take a lot of memory, do we want to
+// hash it? But that can be slow...
+// extracting this string from a typical element takes ~3 microsec, where
+// doing a simple hash ala https://stackoverflow.com/questions/7616461
+// adds ~15 microsec (nearly all of this is spent in charCodeAt)
+// function hash(s) {
+//   var h = 0;
+//   for (var i = 0; i < s.length; i++) {
+//     h = (((h << 5) - h) + s.charCodeAt(i)) | 0; // codePointAt?
+//   }
+//   return h;
+// }
+function nodeHash(node) {
```

Comment:
> Do you have any benchmark results comparing old and new Drawing.bBox?

I don't have any explicit benchmarks, but putting in a few `console.log` statements you can see that previously during redraws (and especially interactions like dragging axis ends) we were only actually using the cache in a few cases, mostly we were redoing the `cloneNode` and `getBoundingClientRect`. Now we hardly ever need to do that. The key is that managing `data-bb` so we can safely use the cache is really hard and we weren't doing it well. This way there's nothing to manage, and `nodeHash` is very fast (as long as I don't actually hash the resulting string 🤔 ).

> 10000 text nodes on one page sounds a bit much.

Yes, perhaps I could drop the limit a bit, but two things to note:
- I'm not storing text nodes, just a lookup object of `{nodeHash: bBox}` where `bBox` is a plain object, my main concern is that `nodeHash` itself can be quite a long string, though I guess a `bBox` object of 6 numbers can also take non-negligible memory.
- A large number of text nodes is exactly when this would be most useful, and wiping the cache will kill this whole gain. Perhaps I could take advantage of `Object.keys` being ordered in insertion order (usually? always?) and drop the oldest 50% of cache entries or something if we hit the limit?